### PR TITLE
Refactor POM Operator

### PR DESCRIPTION
**Pull request by carlosu7**
Stop using static methods and move POM Operator related logic from MavenProvider to reduce public visibility of types

### CODE REVIEW

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/FileUtils.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package io.codemodder.plugins.maven.operator;
```
Comment (carlosu7) [inline plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/FileUtils.java:1]: this was created due to this sonarcloud hotspot issue https://sonarcloud.io/project/security_hotspots?id=pixee_codemodder-java&pullRequest=226&resolved=false&types=SECURITY_HOTSPOT&tab=code

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/MavenProvider.java [hunk 2]
```diff
@@ -163,128 +160,104 @@ private DependencyUpdateResult updateDependenciesInternal(
...
+      LOG.error("Failed to write pom", exc);
+      erroredFiles.add(path);
+    }
+  }
+
+  private URI getPomFileURI(final POMDocument aPomFile) {
+    try {
+      return aPomFile.getPomPath().toURI();
+    } catch (URISyntaxException ex) {
+      ex.printStackTrace();
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/MavenProvider.java:256]:
```suggestion
LOG.error("Unexpected problem getting pom URI", e);
```

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMOperator.java [hunk 3]
```diff
@@ -1,15 +1,89 @@
...
+   * Modifies and retrieves the ProjectModel with a new dependency.
+   *
+   * @param newDependencyGAV The new DependencyGAV to add to the POM.
+   * @return The modified ProjectModel, or null if the modification was unsuccessful.
+   * @throws XMLStreamException If an error occurs during XML stream processing.
+   * @throws URISyntaxException If there is an issue with the URI syntax.
+   * @throws IOException If an I/O error occurs.
+   * @throws DocumentException If an error occurs while parsing the document.
+   */
+  public ProjectModel modifyAndGetProjectModel(final DependencyGAV newDependencyGAV)
```
Comment (nahsra) [inline plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMOperator.java:44]: The name here: `modifyAndGetProjectModel`

Can it be just `addDependency`? Isn't that all we're doing? We happen to return a `ProjectModel`, but it seems like it should be a simpler name.

**Review by nahsra — APPROVED** [review]
Approving, but consider name change

File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMOperator.java (hunk 3 above)
Comment (carlosu7) [reply plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/POMOperator.java:44]: nvm... add dependency works

**Comment by carlosu7** [comment]
sonarcloud is reporting to log the exception but it is been logged... merging this PR

**Pull request merged**